common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -293,4 +294,45 @@ public static String getJsonPath(String jsonPathWithSource) {
         // Extract the substring from the startIndex to the end of the input string
         return (startIndex != -1) ? jsonPathWithSource.substring(startIndex) : jsonPathWithSource;
     }
+
+    /**
+     * Checks if the given input string matches the dot path format.
+     *
+     * <p>The dot path format is a string that consists of one or more word characters
+     * (letters, digits, or underscores) separated by dots. The string can optionally
+     * start or end with a dot followed by one or more word characters.
+     *
+     * <p>Examples of valid dot path format strings:
+     * <ul>
+     *     <li>"foo"</li>
+     *     <li>"foo.bar"</li>
+     *     <li>"foo.bar.baz"</li>
+     *     <li>"foo.bar.baz.qux"</li>
+     *     <li>".foo"</li>
+     *     <li>"foo."</li>
+     *     <li>".foo.bar"</li>
+     * </ul>
+     *
+     * <p>Examples of invalid dot path format strings:
+     * <ul>
+     *     <li>"foo..bar"</li>
+     *     <li>"."</li>
+     *     <li>".."</li>
+     * </ul>
+     *
+     * @param input the input string to be checked
+     * @return true if the input string matches the dot path format, false otherwise
+     */
+    public static boolean isValidJSONPath(String input) {
+        if (input == null || input.equals("")) {
+            return false;
+        }
+        try {
+            JsonPath.compile(input); // This will throw an exception if the path is invalid
```

Comment:
If we are checking only dot path format, we can easily do:

```
    // Regular expression for a valid dot path (letters, digits, underscores, separated by dots)
    String regex = "^\\.?\\w+(\\.\\w+)*\\.?$";
    return input.matches(regex);
```

Reply:
I tried to use regex function earlier but in a deeper thought, I think we need to support all kinds of json path. Some of the json path have some nice functions too, like 'length()', we should support and confirm this is a valid json path. Checkout the function here https://github.com/json-path/JsonPath